src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -100,6 +112,19 @@ const RetrieveInstances::LoadProjectsAndInstancesResult kInitialTestDataWithProj
     std::optional<Project>{kTestProject1},          /* project_of_instances */
 };
 
+std::optional<Project> GetQSettingsProject() {
+  QSettings settings;
+
+  std::optional<Project> project;
+  if (settings.contains(kSelectedProjectIdKey)) {
+    project = Project{
+        settings.value(kSelectedProjectDisplayNameKey).toString() /* display_name */,
+        settings.value(kSelectedProjectIdKey).toString() /* id */
+    };
+  }
+  return project;
+}
```

Comment:
I would consider putting `Set/GetQSettingsProject` into a separate file instead of duplicating the code here.

Reply:
done